src/services/withPath.ts

```diff
@@ -0,0 +1,13 @@
+import { isUrl, Url } from '@/types/url'
+
+export function withPath(url: Url, path: string): Url
+export function withPath(url: string, path: string): Url
+export function withPath(url: string, path: string): Url {
+  const cleanPath = path.replace(/^\/*/, '')
+
+  if (isUrl(url)) {
+    return `${url}${cleanPath}`
+  }
+
+  return `/${url}${cleanPath}`
```

Comment:
Doesn't clean path never have a slash at the beginning so 8-10 won't ever do anything right?

Reply:
cleaning the path doesn't have any effect on the `url` property check on line 8

Reply:
Yeah I totally read that wrong. Carry on